### PR TITLE
feat(confirmation-dialog): Restyle confirmation dialog + add wiggle a…

### PR DESCRIPTION
**Pull request by SirTroyer**
Implement changes suggested in #922 to make the confirmation dialog more prominent.

### <strong>Pull Request</strong>

<hr>

Following changes should are done:

- [x] make it darker
- [x] move buttons to the left (more consistent with our button positioning guidelines + on larger screens the button can be too far away from the content)
- [x] do not dock the confirmation dialog directly to the edge of the screen but give a little margin
- [x] add shadow to make it pop from the content above
- [x] text should be placed above the buttons making the confirmation dialog higher and therefore also improve visibility
- [x] provide option to make the confirmation dialog wiggle (future behaviour for the screens will be the following: when the user tries to navigate away from the screen with unsaved changes we'd like to stay on the page and wait for the user to do either save or discard and therefore draw the users attention to the confirmation dialog with a wiggle animation also indicating that what the user anticipated to do is invalid)

#### Type of PR

Feature (non-breaking change which adds functionality)

### CODE REVIEW

File: libs/barista-components/confirmation-dialog/src/confirmation-dialog.spec.ts [hunk 1]
```diff
@@ -146,10 +141,28 @@ describe('ConfirmationDialogComponent', () => {
...
+      fixture.detectChanges();
+
+      // wiggle again
+      componentInstance.dialog.wiggle();
+      expect(componentInstance.dialog._wiggleState).toEqual(true);
+      fixture.detectChanges();
+      tick();
+      expect(componentInstance.dialog._wiggleState).toEqual(false);
+    }));
   });
```
Comment (ffriedl89) [inline libs/barista-components/confirmation-dialog/src/confirmation-dialog.spec.ts:167]: A testcase where we check whether a change in the viewport resizer changes the left position would be nice.

**Comment by thomaspink** [comment]
@SirTroyer Thank you for the PR. Looks great at a first glance. Will start reviewing it soon. :)

**Comment by ffriedl89** [comment]
@SirTroyer  And you need to sign the cla agreement since this is your first contribution. See the comment and help from the bot above.

**Comment by ffriedl89** [comment]
FYI: We are investigating whether we can test the viewporter resizer and position interaction in jest.

File: libs/barista-components/confirmation-dialog/src/confirmation-dialog.ts [hunk 2]
```diff
@@ -177,6 +198,12 @@ export class DtConfirmationDialog
     this.state = null;
     this._overlayRef.dispose();
     this._stateChildrenSubscription.unsubscribe();
+    this._viewportChangesSubscription.unsubscribe();
+  }
+
+  /** Triggers a wiggle animation on the confirmation dialog */
+  wiggle(): void {
+    this._wiggleState = true;
   }
```
Comment (ffriedl89) [inline libs/barista-components/confirmation-dialog/src/confirmation-dialog.ts:207]: After rethinking this during finalizing this PR - is a more generic function name like `focusAttention()` better - at some point we might decide to switch to another animation. Then wiggle is tightly coupled to the animation used.

Comment (ffriedl89) [reply libs/barista-components/confirmation-dialog/src/confirmation-dialog.ts:207]: I've renamed the method to focusAttention and added the entry to the readme

**Comment by tomheller** [comment]
There was still `wiggle` used in the devApp. I have corrected and changed it to `focusAttention`

**Pull request merged**